hooks/open-telemetry/README.md

```diff
@@ -1,38 +1,92 @@
 # OpenTelemetry Hook
 
-### Requirements
+## Requirements
 
 - open-feature/go-sdk >= v1.3.0
 
 ## Usage
 
-For this hook to function correctly a global `TracerProvider` must be set, an example of how to do this can be found below.
+## Metric hook
 
-The `open telemetry hook` taps into the after and error methods of the hook lifecycle to write `events` and `attributes` to an existing `span`.
+This hook performs metric collection by tapping into various hook stages. Given below are the metrics are extracted 
+by this hook,
+
+- `feature_flag.evaluation_requests_total`
+- `feature_flag.evaluation_success_total`
+- `feature_flag.evaluation_error_total`
+- `feature_flag.evaluation_active_count`
+
+### Options
+
+#### WithFlagMetadataDimensions 
+
+This constructor option allows to configure dimension descriptions to be extracted from `openfeature.FlagMetadata`. 
+If present, these dimension will be added to the `feature_flag.evaluation_success_total` metric.
+
+Note that, this configuration must be carefully coordinated with `Provider` as missing dimensions results in hook 
+evaluation error.
+
+Example usage,
+
+```go
+NewMetricsHook(reader,
+    WithFlagMetadataDimensions(
+        DimensionDescription{
+            Key:  "scope",
+            Type: String,
+        }))
+```
+
+### Example
+
+```go
+// Reader must be configured and injected based from application level
+var reader metric.Reader
+        
+// Derive metric hook from reader
+metricsHook, _ := hooks.NewMetricsHook(reader)
+if err != nil {
+    return err
+}
+
+// Register OpenFeature API level hooks
+openfeature.AddHooks(metricsHook)
+```
+
+## Span hook
+
+For this hook to function correctly a global `TracerProvider` must be set, an example of how to do this can be found
+below.
```

Comment:
Is your IDE automatically adding line breaks based on the sentence length?

Reply:
Yeah, this is from formatting. However, this does not impact the final rendering, see here - https://github.com/open-feature/go-sdk-contrib/blob/30035447e7d114cf2052fed669897cc57f67154d/hooks/open-telemetry/README.md 

IMO, it's good to have a character limit per line and add a line break. However, in .MD rendering this does not necessarily convert to a new line

Reply:
Adding a new line is totally fine but it would make providing feedback easier if it was split per sentence.